tsconfig.json

```diff
@@ -4,8 +4,9 @@
     "lib": [
       "ES2021"
     ],
-    "module": "commonjs",
-    "outDir": "dist",
+    "module": "Node16",
+    "moduleResolution": "Node16",
+    "outDir": "dist_module",
```

Comment:
this is slightly bikeshed-y, but since `package.json` and `tsconfig.json` now express that the "primary" export is esm, maybe `dist/` should be the destination for esm files (since it's more concise and "primary") and cjs should be emitted to something like `dist-cjs/`? i don't feel strongly about this but it might be a little more consistent.

Reply:
Thank you. I don't have any strong opinion, so I adjusted the folders in the latest commit your preference :)